andela/valinor-ah-backend
# #161291003 Implement functionality to send verification email to users

#### What does this PR do?
Have user receive confirmation email on signup.

#### Description of Task to be completed?
- Have an endpoint call the function to send confirmation email.
- Have an endpoint to verify the user when they click the link in confirmation email.

#### How should this be manually tested?
- Clone the repo.
- Cd into it then git checkout into the branch `feature/161291003/email-verification`
- Run `npm install` to install necessary packages.
- Run `npm start:dev` to start the app using nodemon.
- Use postman to test the route `POST: http://localhost:3000/api/users/sendVerification` with a request body containing a JSON object of the following schema:

```
{
	"email": "youremail@example.com",
	"username": "yourusername”
}
```
- Copy the `verificationToken` from the response 
- Use it to test the route `GET http://localhost:3000/api/users/verify?token={verificationToken}`

#### Any background context you want to provide?
The `POST: host/api/users/sendVerification` route is just to test the sendVerification function, it should be refactored into the sign up route.

#### What are the relevant pivotal tracker stories?
#161291003(https://www.pivotaltracker.com/story/show/161291003)

#### Screenshots (if appropriate)
<img width="1129" alt="screen shot 2018-10-29 at 10 14 44 am" src="https://user-images.githubusercontent.com/31810086/47640307-969e4480-db63-11e8-9fde-817dff854b65.png">
<img width="1130" alt="screen shot 2018-10-29 at 10 15 54 am" src="https://user-images.githubusercontent.com/31810086/47640345-ad449b80-db63-11e8-8943-202080689136.png">


#### Questions:
N/A
